src/main/java/org/testng/internal/annotations/IgnoreListener.java

```diff
@@ -21,14 +21,28 @@ public void transform(ITestAnnotation annotation, Class testClass, Constructor t
             typedTestClass = testMethod.getDeclaringClass();
         }
         if (typedTestClass != null) {
-            ignoreTest(annotation, typedTestClass.getAnnotation(Ignore.class));
+            ignoreTest(annotation, findAnnotation(typedTestClass));
             Package testPackage = typedTestClass.getPackage();
             if (testPackage != null) {
                 ignoreTest(annotation, findAnnotation(testPackage));
             }
         }
     }
 
+    private static Ignore findAnnotation(Class<?> typedTestClass) {
```

Comment:
Maybe we already have something similar in an utils class (ReflexionUtils?). Or you can move this function there.

Reply:
Fixed.